src/stream.cpp

```diff
@@ -240,28 +301,19 @@ namespace oxen::quic
 
         auto sz = size();
 
+        if (not _is_writing and _unacked_size == 0)
```

Comment:
I think that this should be:
```suggestion
        if (not _is_writing and user_buffers.empty())
```
because I think it's possible to have *no* unacked data, but still have some buffered data that hasn't been sent yet (perhaps because sending is paused, or just because of very unlucky network conditions).  If there is something in user_buffers, however, then that implies that there is something either unsent or unacked (or both).